docroot/profiles/custom/cgov_site/modules/custom/cgov_yaml_content/src/Service/YamlEntityEmbedProcessor.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace Drupal\cgov_yaml_content\Service;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\TypedData\Exception\MissingDataException;
+use DOMWrap\Document;
+
+/**
+ * Helper service for YAML content loading for <drupal-entity> tags.
+ *
+ * This service will replace <drupal-entity> UUID attributes with the UUIDs of
+ * content that had been previously loaded in our content loading.
+ *
+ * NOTE: For sanity, this only supports ContentEntities as they can have fields.
+ */
+class YamlEntityEmbedProcessor {
+
+  /**
+   * Drupal Entity Type Manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Create new Event Subscriber class.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   Drupal Entity Type Manager.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entityTypeManager
+  ) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * This is the main entry point that will process an entity.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity to be processed.
+   */
+  public function processEntity(ContentEntityInterface $entity) {
+
+    // We only need to check text_long fields as those are the ones that can
+    // hold HTML, and thus only the ones to contain <drupal-entity> tags.
+    foreach ($entity->getFields() as $field) {
+      if ($field->getFieldDefinition()->getType() === 'text_long') {
+        $this->handleEmbedForTextLong($field);
+      }
+    }
+  }
+
+  /**
+   * Function for processing <drupal-embed> tags in a text_long field.
+   *
+   * @param \Drupal\Core\Field\FieldItemListInterface $field
+   *   The Field to process.
+   */
+  public function handleEmbedForTextLong(FieldItemListInterface $field) {
```

Comment:
A style question -- what's the reason for this method being declared public? I don't think it's a problem per se, but it's only used by this class.  Not even from a unit test.  Not the first place I've seen this in our code base, but it seems like a potential point of confusion for future consumers of the class.

Reply:
In case/when we add tests for this at some point. It is a fair point, but I was just going to leave them public for now.

Reply:
👍